module-4/backend/week_three.md

```diff
@@ -4,8 +4,54 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 
 Note: When you're done, submit a PR. 
 
-TBD  
+1. At a high level, what is Node?
+
+Node is a platform for running server-side Javascript. It runs on v8.
+
+2. What is Express? Why is Express similar to in the Ruby world?
+
+Express is a lightweight Javascript framework, similar to Ruby's Sinatra.
+
+3. How do we setup a route when creating an API with Node and Express? Please provide a code snippet.
+
+```
+app.get(url, (request, response) => {
+  response.json(stuff)
+)}
+```
+PS. I admit that I had to look at my notes to remember the syntax for this. I haven't memorized it yet. I'm hoping using Express for my capstone will help commit this stuff to my working memory.
+
+4. What do we use Knex for?
+
+To interface with a database in Node. We've used it for postgresql... I wonder if it's the same for other databases?
+
+5. How could you organize your code to follow the MVC design pattern in your Quantified Self project?
+
+Move routes into their own files in a controller folder. Move logic out of the routes and into files in a model folder. The view is already separated out into its own repo.
+
+6. How do you execute raw SQL in node?
+
+knex.raw("INSERT SQL STUFF HERE")
+
+7. What are some advantages to having your front end and back end code live in separate applications? What are some disadvantages?
+
+Organizations do this so that they can run apps on different platforms without having to reinvent the backend of the app each time. It provides a pretty clear separation between the Models/Controllers and the View of MVC apps. Not sure what the disadvantages are. To me, the disadvantage is that it seems harder and less familiar, but that seems like a short-sighted thing to list as a disadvantage.
```

Comment:
Good! It also provides bigger challenges for things like SEO and authentication.